File: backend/api/manager/validation_manager.py
import re

def validate(input: str) -> str :
  validInput = _filter(_check_length(input))
  return validInput
  
def _filter(input: str) -> str :
  whitelist = "[^\\w\\söÖäÄüÜ0-9?!():&.,;#]"
  validInput = re.sub(whitelist, "", input)
  return validInput

def _check_length(input: str) -> str :
  if len(input) <= 300:
    return input
  return input[0:301]
  
# if __name__ == "__main__":
#   input = "11 hello there <script> lsldsl </script> ^^!"
#   val = validate(input)
#   print(val)

File: frontend/src/manager/ValidationManager.ts
export class ValidationManager {
  validate(input: string): string {
    let validInput = this.escape(input);
    validInput = this.filter(validInput);
    return validInput;
  }

  /**
   * HTML Character Entities (case sensitive -> g):
   *  & 	&amp; 	
   *  < 	&lt; 	
   *  > 	&gt; 	
   *  " 	&quot; 	
   *  ' 	&apos; 	
   * @param input
   * @returns escaped / sanitized string
   */
  private escape(input: string): string {
    return input
      .replace(/&/g, "&amp;")
      .replace(/</g, "&lt;")
      .replace(/>/g, "&gt;")
      .replace(/"/g, "&quot;")
      .replace(/'/g, "&#x27;")
      .replace(/\//g, "&#x2F;")
      .replace(/\\/g, "&#x5C;")
      .replace(/`/g, "&#96;");
  }

  private filter(input: string): string {
    const whitelist = new RegExp("[^\\w\\söÖäÄüÜ?!():&.,;#]", "g");
    return input.replace(whitelist, "");
  }
}
